R/clean_code.R

```diff
@@ -0,0 +1,151 @@
+utils::globalVariables(
+  c("cnt", "start", "impl", "lifetime", "type", "name")
+)
+
+clean_rust_code <- function(lines) {
+  lines <-
+    lines %>%
+    remove_empty_or_whitespace() %>%
+    fill_block_comments() %>%
+    remove_line_comments() %>%
+    remove_empty_or_whitespace()
+}
+
+remove_empty_or_whitespace <- function(lns) {
+  stringi::stri_subset_regex(lns, "^\\s*$", negate = TRUE)
+}
+
+remove_line_comments <- function(lns) {
+  stringi::stri_replace_first_regex(lns, "//.*$", "")
+}
+
+# Because R does not allow strightforward iteration over
+# scalar strings, determining `/*` and `*/` positions can be challenging.
+# E.g., regex matches 3 `/*` and 3 `*/` in `/*/**/*/`.
+# 1. We find all occurence of `/*` and `*/`.
+# 2. We find non-overlapping `/*` and `*/`.
+# 3. We build pairs of open-close comment delimiters by collapsing nested
+#   comments.
+# 4. We fill in space between remaining delimiters with spaces (simplest way).
+fill_block_comments <- function(lns, fill_with = " ") {
+  lns <- paste(lns, collapse = "\n")
```

Comment:
Let's use `glue` functions consistently instead of `paste()`, `paste0()`, etc. This here would probably be `glue_collapse()`.